src/Component/PlaceOrder.js

```diff
@@ -0,0 +1,145 @@
+import React, { useContext } from "react";
+import "./PlaceOrder.css";
+import { useFormik } from "formik";
+import { CartContext } from "../App";
+import { useHistory } from "react-router-dom";
+import SignIn from "./SignIn";
+
+function PlaceOrder() {
+  const context = useContext(CartContext);
+  const history = useHistory();
+  const initialValues = {
+    name: "",
+    email: context.cartUserState ? context.cartUserState.email : "",
+    mob: "",
+    addr: "",
+    creditNo: "",
+  };
+  const onSubmit = (values) => {
+    alert("Order placed Successfully!");
+    context.dispatchState({
+      type: "clearCart",
+    });
+    history.push("/");
+  };
+  const validate = (values) => {
+    let errors = {};
+    if (!values.name) {
+      errors.name = "Name is Required!";
+    } else if (
+      !/^[a-zA-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$/.test(values.name)
+    ) {
+      errors.name = "Please enter valid Name!";
+    }
+
+    if (!values.email) {
+      errors.email = "Email is Required";
+    } else if (
+      !/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/.test(
+        values.email
+      )
+    ) {
+      errors.email = "Please enter valid email!";
+    }
+    if (!values.mob) {
+      errors.mob = "Mobile No is Required!";
+    } else if (isNaN(values.mob)) {
+      errors.mob = "Please enter valid Mobile No!";
+    } else if (values.mob.length !== 10) {
+      errors.mob = "Mobile No must contain 10 digits!";
+    }
+    if (!values.addr) errors.addr = "Address is Required!";
+    if (!values.creditNo) {
+      errors.creditNo = "Credit Card No is Required!";
+    } else if (isNaN(values.creditNo)) {
+      errors.creditNo = "Please enter valid Credit Card No!";
+    } else if (values.creditNo.length !== 16) {
+      errors.creditNo = "Credit Card No must contain 16 digits!";
+    }
+    return errors;
+  };
+
+  const formik = useFormik({
+    initialValues,
+    onSubmit,
+    validate,
+  });
+
+  return (
+    <form onSubmit={formik.handleSubmit}>
```

Comment:
I would not prefer to use library for such a small requirments. Developer ease might cause unwanted effects. formik is good but I won't advise adding 500KB to bundle size for such a small requirements.

Reply:
Yes, true that is... I will try to handle form validations without any library